tests/Integration/Momento.Sdk.Tests/TopicTest.cs

```diff
@@ -0,0 +1,244 @@
+    public async Task PublishAsync_PublishNullString_IsError()
+    {
+        var response = await topicClient.PublishAsync(cacheName, "topic", (string)null!);
+        Assert.True(response is TopicPublishResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, ((TopicPublishResponse.Error)response).ErrorCode);
+    }
+
+    [Fact]
+    public async Task PublishAsync_BadCacheNameString_IsError()
+    {
+        var response = await topicClient.PublishAsync("fake-" + cacheName, "topic", "value");
+        Assert.True(response is TopicPublishResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.NOT_FOUND_ERROR, ((TopicPublishResponse.Error)response).ErrorCode);
+    }
+
+    [Theory]
+    [InlineData(null, "topic")]
+    [InlineData("cache", null)]
+    public async Task SubscribeAsync_NullChecks_IsError(string badCacheName, string badTopicName)
+    {
+        var response = await topicClient.SubscribeAsync(badCacheName, badTopicName);
+        Assert.True(response is TopicSubscribeResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, ((TopicSubscribeResponse.Error)response).ErrorCode);
+    }
+
+    [Fact]
+    public async Task PublishAndSubscribe_ByteArray_Succeeds()
+    {
+        const string topicName = "topic_bytes";
+        var skipMessage = new byte[] { 0x00 };
+        var valuesToSend = new List<byte[]>
+        {
+            new byte[] { 0x01 },
+            new byte[] { 0x02 },
+            new byte[] { 0x03 },
+            new byte[] { 0x04 },
+            new byte[] { 0x05 }
+        };
+
+        var subscribeResponse = await topicClient.SubscribeAsync(cacheName, topicName);
+        Assert.True(subscribeResponse is TopicSubscribeResponse.Subscription,
+            $"Unexpected response: {subscribeResponse}");
+        var subscription = (TopicSubscribeResponse.Subscription)subscribeResponse;
+        
+        var delayTask = Task.Delay(5000);
+
+        var testTask = Task.Run(async () =>
+        {
+            var messageCount = 0;
+            await foreach (var message in subscription)
+            {
+                Assert.NotNull(message);
+                Assert.True(message is TopicMessage.Item, $"Unexpected message: {message}");
+                if (((TopicMessage.Item)message).ValueByteArray!.SequenceEqual(skipMessage)) continue;
+                
+                Assert.Equal(valuesToSend[messageCount], ((TopicMessage.Item)message).ValueByteArray);
+
+                messageCount++;
+                if (messageCount == valuesToSend.Count)
+                {
+                    break;
+                }
+            }
+
+            subscription.Dispose();
+            return messageCount;
+        });
+        
+        // Send a few values to skip over to ensure the subscription received the messages under test
+        for (var i = 0; i < 5; i++)
```

Comment:
This feels hacky but I was having a hard time getting the subscription to consistently receive the first message.

Reply:
Have you tried just putting a sleep here before you start publishing?